packages/sol-doc/src/solidity_doc_generator.ts

```diff
@@ -0,0 +1,284 @@
+            for (const contractName of contractNames) {
+                const compiledContract = contractNameToOutput[contractName];
+                if (_.isUndefined(compiledContract.abi)) {
+                    throw new Error('compiled contract did not contain ABI output');
+                }
+                doc[contractName] = _genDocSection(compiledContract);
+            }
+        }
+    }
+
+    return doc;
+}
+
+function _makeCompilerOptions(contractsToCompile: string[], contractsDir: string): CompilerOptions {
+    const compilerOptions: CompilerOptions = {
+        contractsDir,
+        contracts: '*',
+        compilerSettings: {
+            outputSelection: {
+                ['*']: {
+                    ['*']: ['abi', 'devdoc'],
+                },
+            },
+        },
+    };
+
+    const shouldOverrideCatchAllContractsConfig = !_.isUndefined(contractsToCompile);
+    if (shouldOverrideCatchAllContractsConfig) {
+        compilerOptions.contracts = contractsToCompile;
+    }
+
+    return compilerOptions;
+}
+
+function _genDocSection(compiledContract: StandardContractOutput): DocSection {
+    const docSection: DocSection = {
+        comment: _.isUndefined(compiledContract.devdoc) ? '' : compiledContract.devdoc.title,
+        constructors: [],
+        methods: [],
+        properties: [],
+        types: [],
+        functions: [],
+        events: [],
+    };
+
+    for (const abiDefinition of compiledContract.abi) {
+        switch (abiDefinition.type) {
+            case 'constructor':
+                docSection.constructors.push(_genConstructorDoc(abiDefinition, compiledContract.devdoc));
+                break;
+            case 'event':
+                (docSection.events as Event[]).push(_genEventDoc(abiDefinition));
+                // note that we're not sending devdoc to _genEventDoc().
+                // that's because the type of the events array doesn't have any fields for documentation!
+                break;
+            case 'function':
+                docSection.methods.push(_genMethodDoc(abiDefinition, compiledContract.devdoc));
+                break;
+            default:
+                throw new Error(`unknown and unsupported AbiDefinition type '${abiDefinition.type}'`);
+        }
+    }
+
+    return docSection;
+}
+
+function _genConstructorDoc(abiDefinition: ConstructorAbi, devdocIfExists: DevdocOutput | undefined): SolidityMethod {
+    const { parameters, methodSignature } = _genMethodParamsDoc(
+        '', // TODO: update depending on how constructors are keyed in devdoc
+        abiDefinition.inputs,
+        devdocIfExists,
+    );
+
+    const comment = _devdocMethodDetailsIfExist(methodSignature, devdocIfExists);
+    // TODO: figure out why devdoc'd constructors don't get output by solc
+
+    const constructorDoc: SolidityMethod = {
+        isConstructor: true,
+        name: '', // sad we have to specify this
+        callPath: '', // TODO: wtf is this?
```

Comment:
`callPath` is used to display methods at `Exchange.fillOrder(` rather then `fillOrder(`, making the doc signatures more copy-pastable. There is debate as to whether we should do this, and I'm open to arguments either way.

Reply:
I don't really understand... Certainly not enough to be making arguments one way or the other...

I just removed the TODO and left the empty string.  If you'd like to see something else here, please let me know.